### PR TITLE
[back/front] Add States to Location (#2385)

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* Add a new location entity type **Administrative-Area**.
* Delete **convertEntityTypeToStixType** from `schemaUtils.js` because `stix-converter.ts` has an efficient **convertTypeToStixType** method

### Related issues

* [issue/2385](https://github.com/OpenCTI-Platform/opencti/issues/2385)

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

Backend part only

### CODE REVIEW

File: opencti-platform/opencti-graphql/config/schema/opencti.graphql [hunk 1]
```diff
@@ -5164,7 +5158,7 @@ interface Location {
 input LocationAddInput {
   stix_id: StixId
   x_opencti_stix_ids: [StixId]
-  type: LocationType!
+  type: String!
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/config/schema/opencti.graphql:5252]: Why this change ?

File: opencti-platform/opencti-graphql/src/domain/location.js [hunk 2]
```diff
@@ -13,7 +14,7 @@ export const findById = async (context, user, locationId) => {
 export const findAll = async (context, user, args) => {
   let types = [];
   if (args.types && args.types.length > 0) {
-    types = filter((type) => isStixDomainObjectLocation(type), args.types);
+    types = R.filter((type) => isStixDomainObjectLocation(type), args.types);
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/domain/location.js]: Please use standard JS filter function instead of Ramda

File: opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { notify } from '../../database/redis';
+import { BUS_TOPICS } from '../../config/conf';
+import { ABSTRACT_STIX_DOMAIN_OBJECT } from '../../schema/general';
+import type { AdministrativeAreaAddInput, QueryAdministrativeAreasArgs } from '../../generated/graphql';
+import { listEntitiesPaginated, storeLoadById } from '../../database/middleware-loader';
+import { BasicStoreEntityAdministrativeArea, ENTITY_TYPE_LOCATION_ADMINISTRATIVE_AREA } from './administrativeArea-types';
+import { RELATION_LOCATED_AT } from '../../schema/stixCoreRelationship';
+import { ENTITY_TYPE_LOCATION_COUNTRY } from '../../schema/stixDomainObject';
+
+export const findById = (context: AuthContext, user: AuthUser, administrativeAreaId: string): BasicStoreEntityAdministrativeArea => {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts]: There is a defined type for this method, please use it

File: opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  return storeLoadById(context, user, administrativeAreaId, ENTITY_TYPE_LOCATION_ADMINISTRATIVE_AREA) as unknown as BasicStoreEntityAdministrativeArea;
+};
+
+export const findAll = (context: AuthContext, user: AuthUser, opts: QueryAdministrativeAreasArgs) => {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts]: same

File: opencti-platform/opencti-graphql/config/schema/opencti.graphql (hunk 1 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/config/schema/opencti.graphql:5252]: LocationType has been removed from schema
the check on type is done in` location.js` > `addLocation`

File: opencti-platform/opencti-graphql/src/domain/location.js (hunk 2 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/src/domain/location.js]: ok

File: opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts (hunk 4 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea-domain.ts]: I found the type for `findAll` but I can't use it


Capture d'écran_20230113_145144

File: opencti-platform/opencti-graphql/src/schema/stixDomainObject.ts [hunk 5]
```diff
@@ -97,7 +97,8 @@ const STIX_DOMAIN_OBJECT_LOCATIONS: Array<string> = [
 ];
 schemaTypes.register(ENTITY_TYPE_LOCATION, STIX_DOMAIN_OBJECT_LOCATIONS);
 export const isStixDomainObjectLocation = (type: string): boolean => {
-  return R.includes(type, STIX_DOMAIN_OBJECT_LOCATIONS) || type === ENTITY_TYPE_LOCATION;
+  const locationEntities = schemaTypes.get(ENTITY_TYPE_LOCATION);
+  return R.includes(type, locationEntities) || type === ENTITY_TYPE_LOCATION;
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/schema/stixDomainObject.ts]: You can replace Ramda here by https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

File: opencti-platform/opencti-graphql/src/schema/schemaUtils.js [hunk 6]
```diff
@@ -55,26 +47,6 @@ export const shortHash = (element) => {
   return hash.slice(0, 8);
 };
 
-export const convertEntityTypeToStixType = (type) => {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/schema/schemaUtils.js:58]: Top !

File: opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea.graphql [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    key: [AdministrativeAreasFilter!]!
+    values: [String]
+    operator: String
+    filterMode: FilterMode
+}
+
+# Relay connections
+type AdministrativeAreaConnection {
+    pageInfo: PageInfo!
+    edges: [AdministrativeAreaEdge]
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/administrativeArea/administrativeArea.graphql]: Should be [AdministrativeAreaEdge!]! because if you have a connection, you will have elements in it

File: opencti-platform/opencti-graphql/config/schema/opencti.graphql (hunk 1 above)
Comment (richard-julien) [reply opencti-platform/opencti-graphql/config/schema/opencti.graphql:5252]: Done because of graphQL enum not supporting the "-" character...
